dpe/src/context.rs

```diff
@@ -150,70 +150,175 @@ pub(crate) struct ChildToRootIter<'a> {
+            context.state = ContextState::Active;
         }
 
         let mut count = 0;
-        let leaf_index = chain_indeces[chain_indeces.len() - 1];
+        let leaf_index = CHAIN_INDICES[CHAIN_INDICES.len() - 1];
 
-        for (answer, context) in chain_indeces
+        for (answer, status) in CHAIN_INDICES
             .iter()
             .rev()
-            .zip(ChildToRootIter::new(leaf_index, &contexts).unwrap())
+            .zip(ChildToRootIter::new(leaf_index, &contexts))
         {
-            assert_eq!(*answer, context.tag as usize);
+            assert_eq!(*answer, status.unwrap().tag as usize);
             count += 1;
         }
 
         // Check we didn't accidentally skip any.
-        assert_eq!(chain_indeces.len(), count);
+        assert_eq!(CHAIN_INDICES.len(), count);
     }
+
+    #[test]
+    fn test_child_to_root_overflow() {
+        const INITIALIZER_CONTEXT: Context = Context::new();
+        let mut contexts = [INITIALIZER_CONTEXT; 2];
+
+        // Create circular relationship.
+        contexts[0].parent_idx = 1;
+        contexts[0].state = ContextState::Active;
+        contexts[1].parent_idx = 0;
+        contexts[1].state = ContextState::Active;
+
+        let mut iter = ChildToRootIter::new(0, &contexts);
+        for _ in 0..MAX_HANDLES {
+            iter.next().unwrap().unwrap();
+        }
+
+        assert_eq!(DpeErrorCode::MaxTcis, iter.next().unwrap().err().unwrap());
+    }
+
+    #[test]
+    fn test_child_to_root_check_parent_and_state() {
+        const INITIALIZER_CONTEXT: Context = Context::new();
+        let mut contexts = [INITIALIZER_CONTEXT];
+        contexts[0].state = ContextState::Retired;
+        contexts[0].parent_idx = MAX_HANDLES as u8;
+
+        // Above upper bound of handles.
+        let mut iter = ChildToRootIter::new(0, &contexts);
+        assert_eq!(
+            DpeErrorCode::InternalError,
+            iter.next().unwrap().err().unwrap()
+        );
+
+        // Inactive.
+        contexts[0].state = ContextState::Inactive;
+        contexts[0].parent_idx = 0;
+        let mut iter = ChildToRootIter::new(0, &contexts);
+        assert_eq!(
+            DpeErrorCode::InternalError,
+            iter.next().unwrap().err().unwrap()
+        );
+
+        // Retired.
+        contexts[0].state = ContextState::Retired;
+        let mut iter = ChildToRootIter::new(0, &contexts);
+        assert!(iter.next().unwrap().is_ok());
+
+        // Active and upper bound of handles.
+        contexts[0].state = ContextState::Active;
+        contexts[0].parent_idx = (MAX_HANDLES - 1) as u8;
+        let mut iter = ChildToRootIter::new(0, &contexts);
+        assert!(iter.next().unwrap().is_ok());
+
+        // Root index.
+        contexts[0].parent_idx = Context::ROOT_INDEX as u8;
+        let mut iter = ChildToRootIter::new(0, &contexts);
+        assert!(iter.next().unwrap().is_ok());
+    }
+
+    const CHAIN_INDICES: [usize; MAX_HANDLES] = [
```

Comment:
Could you add a comment here about what this chain of indices is meant to test?

Reply:
Added a comment with some more description of how it is used.